pennylane/ops/qubit/observables.py

```diff
@@ -181,6 +181,8 @@ class SparseHamiltonian(Observable):
     grad_method = None
 
     def __init__(self, H, wires=None, do_queue=True, id=None):
+        if not isinstance(H, coo_matrix):
+            raise TypeError("Observable must be a scipy sparse coo_matrix.")
         super().__init__(H, wires=wires, do_queue=do_queue, id=id)
```

Comment:
@josh146 for visibility: when tidying up params, `super` should get nothing and `H` should be stored as a `hyperparameter`. Shall I do it in this PR?

Reply:
I've already done it in #2049 :)